### ISSUE TITLE
Support generic types with trait bounds

### ISSUE THREAD

**Issue by Geal** [BrynCooke/buildstructor#60]
When using buildstructor on a generic type, I get the following error:

```
error[E0107]: missing generics for struct `services::ExecutionResponse`
   --> apollo-router-core/src/services/mod.rs:605:34
    |
605 | impl<T: Stream<Item = Response>> ExecutionResponse<T> {
    |                                  ^^^^^^^^^^^^^^^^^ expected 1 generic argument
    |
note: struct defined here, with 1 generic parameter: `T`
   --> apollo-router-core/src/services/mod.rs:598:12
    |
598 | pub struct ExecutionResponse<T: Stream<Item = Response>> {
    |            ^^^^^^^^^^^^^^^^^ -
help: add missing generic argument
    |
605 | impl<T: Stream<Item = Response>> ExecutionResponse<T><T> {
    |                                  ~~~~~~~~~~~~~~~~~~~~
```

Apparently the error persists even if I specify an explicit type

**Comment by BrynCooke** [comment]
@Geal I suspect this is already fixed. Was this with Buildstructor 0.3?

**Comment by BrynCooke** [comment]
As far as I can tell this is working: https://github.com/BrynCooke/buildstructor/blob/main/tests/buildstructor/pass/generic.rs#L6-L28 

Going to close, but can reopen if there is a specific reproduction.

**Comment by BrynCooke** [comment]
Here is a reproduction:
```
#[buildstructor]
impl Foo<usize> {
    #[builder]
    fn bound3_new(simple: usize) -> Foo<usize> {
        Self { simple }
    }
}
```
It seems to affect specialization.

**Comment by BrynCooke** [comment]
Fixed in 0.3.2
